src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.Streaming.cs

```diff
@@ -275,6 +277,12 @@ private void WriteComponentHtml(int componentId, TextWriter output, bool allowBo
             {
                 if (_httpContext.RequestServices.GetRequiredService<WebAssemblySettingsEmitter>().TryGetSettingsOnce(out var settings))
                 {
+                    if (marker.Type is ComponentMarker.WebAssemblyMarkerType)
+                    {
+                        // Preload WebAssembly assets when using WebAssembly (not Auto) mode
+                        AppendWebAssemblyPreloadHeaders();
+                    }
```

Comment:
Is this unconditionally emitting the headers or is it only doing it when a webassembly component is rendered on to the page?

Reply:
It's executed only when a first WebAssembly component is rendered.